Modules/RPC.cs

```diff
@@ -877,35 +645,248 @@ public static void SetCustomRole(byte targetId, CustomRoles role)
 
         if (PlayerControl.LocalPlayer.PlayerId == targetId) RemoveDisableDevicesPatch.UpdateDisableDevices();
     }
+    private static int Lastrpc = -1;
     public static void SyncRoleSkillReader(MessageReader reader, PlayerControl pc)
     {
-        CustomRoles role = (CustomRoles)reader.ReadPackedInt32();
-        Logger.Info($"Received Sync Role Skill RPC for role {role}", "SyncRoleSkillReader");
+        int RpcInx = Lastrpc;
+
+        try
+        {
+            RpcInx = reader.ReadPackedInt32();
+        }
+        catch { }
+
+        if (RpcInx != -1)
+        {
+            Lastrpc = RpcInx;
+        }
+
+        Logger.Info($"Recieved Rpc for role: {pc.GetRoleClass().GetType().Name} | With Rpcinx?: {(RpcInx != -1 ? RpcInx : false)} | Ishost?: {AmongUsClient.Instance.AmHost} ", "SyncRoleSkillReader");
 
         try
         {
-            switch (role)
+            switch (pc.GetRoleClass())
             {
-                //Crew Roles
-                case CustomRoles.Admirer:
-                    Admirer.ReceiveRPC(reader, false);
+                case Admirer ad when RpcInx == 1:
+                    ad.ReceiveRPC(reader, false);
+                    break;
+                case Admirer ad when RpcInx == 2:
+                    ad.ReceiveRPC(reader, true);
+                    break;
```

Comment:
I think this method with `RpcInx` is much better (since there is no need to add RPC manually)
But this is not necessary to change it

![image](https://github.com/0xDrMoe/TownofHost-Enhanced/assets/104814436/7befb0a7-35c0-48e8-86cf-9d3c85562507)

Reply:
oh I see, well we can do it for every new role. And if I have time to make a new pr to make all of the roles like that..